homeassistant/components/modbus/fan.py

```diff
@@ -0,0 +1,188 @@
+        self._name = config[CONF_NAME]
+        self._slave = config.get(CONF_SLAVE)
+        self._is_on = None
+        self._available = True
+        self._scan_interval = timedelta(seconds=config[CONF_SCAN_INTERVAL])
+        self._address = config[CONF_ADDRESS]
+        if config[CONF_WRITE_TYPE] == CALL_TYPE_COIL:
+            self._write_func = self._hub.write_coil
+            self._command_on = 0x01
+            self._command_off = 0x00
+        else:
+            self._write_func = self._hub.write_register
+            self._command_on = config[CONF_COMMAND_ON]
+            self._command_off = config[CONF_COMMAND_OFF]
+        if CONF_VERIFY in config:
+            self._verify_active = True
+            self._verify_address = config[CONF_VERIFY].get(
+                CONF_ADDRESS, config[CONF_ADDRESS]
+            )
+            self._verify_type = config[CONF_VERIFY].get(
+                CONF_INPUT_TYPE, config[CONF_WRITE_TYPE]
+            )
+            self._state_on = config[CONF_VERIFY].get(CONF_STATE_ON, self._command_on)
+            self._state_off = config[CONF_VERIFY].get(CONF_STATE_OFF, self._command_off)
+
+            if self._verify_type == CALL_TYPE_REGISTER_HOLDING:
+                self._read_func = self._hub.read_holding_registers
+            elif self._verify_type == CALL_TYPE_DISCRETE:
+                self._read_func = self._hub.read_discrete_inputs
+            elif self._verify_type == CALL_TYPE_REGISTER_INPUT:
+                self._read_func = self._hub.read_input_registers
+            else:  # self._verify_type == CALL_TYPE_COIL:
+                self._read_func = self._hub.read_coils
+        else:
+            self._verify_active = False
+
+    async def async_added_to_hass(self):
+        """Handle entity which will be added."""
+        state = await self.async_get_last_state()
+        if state:
+            self._is_on = state.state == STATE_ON
+
+        if self._verify_active:
+            async_track_time_interval(
+                self.hass, lambda arg: self.update(), self._scan_interval
+            )
+
+    @property
+    def is_on(self):
+        """Return true if fan is on."""
+        return self._is_on
+
+    @property
+    def name(self):
+        """Return the name of the fan."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """Return True if entity has to be polled for state."""
+        return False
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    def turn_on(
+        self,
+        speed: str | None = None,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs,
+    ) -> None:
+        """Set fan on."""
+
+        result = self._write_func(self._slave, self._address, self._command_on)
+        if result is False:
+            self._available = False
+            self.schedule_update_ha_state()
+        else:
```

Comment:
We can return here instead and outdent below.
```suggestion
            return
```